public/app/plugins/datasource/prometheus/legend.ts

```diff
@@ -1,7 +1,28 @@
 import { Labels } from '@grafana/data';
 
-/** replace labels in a string.  Used for loki+prometheus legend formats */
+/**
+ * Replace labels in a string.  Used for loki+prometheus legend formats
+ *
+ * For many years, this was implemented as:
+ *   const aliasRegex = /\{\{\s*(.+?)\s*\}\}/g;
+ */
 export function renderLegendFormat(aliasPattern: string, aliasData: Labels): string {
-  const aliasRegex = /\{\{\s*(.+?)\s*\}\}/g;
-  return aliasPattern.replace(aliasRegex, (_, g1) => (aliasData[g1] ? aliasData[g1] : g1));
+  let idx = aliasPattern ? aliasPattern.indexOf('{{') : -1;
+  while (idx >= 0 && aliasData) {
+    let edx = aliasPattern.indexOf('}}', idx + 1);
+    if (idx > edx) {
+      return aliasPattern;
+    }
+
+    const key = aliasPattern.substring(idx + 2, edx).trim();
+    let val = aliasData[key];
+    if (val == null && Object.keys(aliasData).length === 0) {
```

Comment:
Is `==` intended here instead of `===` to take `val === undefined` into account? If yes, is it covered by tests?